src/backend/file_chooser.rs

```diff
@@ -22,6 +26,42 @@ use crate::{
     AppID, WindowIdentifierType,
 };
 
+// FIXME Manually implement Deserialize to catch non null-terminated arrays.
+// FIXME Use OsString directly once
+// https://gitlab.freedesktop.org/dbus/zbus/-/merge_requests/638 is merged.
+/// A file name represented as a null-terminated byte array.
+///
+/// Implements `AsRef<Path>`.
+#[derive(Debug, Default)]
+pub struct FileName(CString);
+
+impl AsRef<Path> for FileName {
+    fn as_ref(&self) -> &Path {
+        let os_str = OsStr::from_bytes(self.0.as_bytes());
+
+        Path::new(os_str)
+    }
+}
+
+impl Type for FileName {
+    fn signature() -> zbus::zvariant::Signature<'static> {
+        <&[u8]>::signature()
+    }
+}
+
+impl<'de> Deserialize<'de> for FileName {
```

Comment:
Please also implement Serialize so that we can  drop stuff like https://github.com/bilelmoussaoui/ashpd/blob/master/src/desktop/file_chooser.rs#L443-L447.

Reply:
In the pr it has serialize, fortunately we don't need a custom serialize.